### PR TITLE
BAH Update to 4142 schema for submission bug

### PR DESCRIPTION
**Pull request by robfelder**
## Description
This PR covers a fix for submitting `treatmentDateRange` the 4142 section of the 526 all claims form.

Related to this story: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/15609

## Testing done
Manual & Unit

## Screenshots
n/a

## Acceptance criteria
- [x] Schema for the UI collects the treatmentDateRange as an object

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

**Note:** the following PR should be merged in before testing this: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/305

### CODE REVIEW

**Review by imeecuison — APPROVED** [review]
Looks good!

**Review by imeecuison — APPROVED** [review]
Looks good!

File: src/applications/disability-benefits/all-claims/pages/privateMedicalRecordsRelease.js [hunk 1]
```diff
@@ -87,7 +90,27 @@ export const uiSchema = {
...
+        type: 'object',
+        required: [
+          'providerFacilityName',
+          'treatmentDateRange',
+          'providerFacilityAddress',
+        ],
+        properties: {
+          providerFacilityName,
+          treatmentDateRange: {
+            type: 'object',
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/pages/privateMedicalRecordsRelease.js]: this isn't supposed to be an 'array'?

**Review by erikphansen — APPROVED** [review]
LGTM, with one question.

File: src/applications/disability-benefits/all-claims/pages/privateMedicalRecordsRelease.js (hunk 1 above)
Comment (robfelder) [reply src/applications/disability-benefits/all-claims/pages/privateMedicalRecordsRelease.js]: It's not. The short story is that the UI isn't supposed to have the ability to add multiple treatmentDateRange's (yet). But the possibility to do it later didn't want to be closed off. So the Schema and API accept the array, with a little transformation.

**Pull request merged**